module_utils/network_lsr/argument_validator.py

```diff
@@ -1958,9 +1958,8 @@ def _ipv6_ignore(connection):
                     idx,
                     "IPv4 needs to be enabled to support IPv4 nameservers.",
                 )
-            if (
-                nameserver["family"] == socket.AF_INET6
-                and connection["ip"]["ipv6_disabled"]
+            if nameserver["family"] == socket.AF_INET6 and (
+                connection["ip"]["ipv6_disabled"] or _ipv6_ignore(connection)
```

Comment:
All these changes are in `ArgValidator.validate_connection_one()`. That function is not intended for such checks.

We have `ArgValidator*.validate()` -- which in particular calls `ArgValidator*._validate_post()`, which is the place for consistency checks that only depend on the values of one the playbook. In particular, `ArgValidator_DictConnection.validate()` verifies (and normalizes) one profile individually, and `ArgValidator_ListConnections.validate()` verifies (and normalizes) the entire list of profiles as a whole.

I guess, `ArgValidator_ListConnections.validate_connection_one()` is a bad name. But that function performs some validation *afterwards*, in particular depending on the provider (`mode`). For example, the `initscripts` provider does not support team. So `ArgValidator*.validate()` (which is independent of the provider/mode) cannot tell whether the playbook is valid for "initscripts". That validation can only happen in a second step: `validate_connection_one()`.

Here all these validation checks are independent of the mode/provider.


I don't mind. You can leave it like this. But in the future, this code should move.